homeassistant/components/miele/binary_sensor.py

```diff
@@ -0,0 +1,283 @@
+            MieleAppliance.WINE_CONDITIONING_UNIT,
+            MieleAppliance.WINE_STORAGE_CONDITIONING_UNIT,
+            MieleAppliance.STEAM_OVEN_MICRO,
+            MieleAppliance.DIALOG_OVEN,
+            MieleAppliance.WINE_CABINET_FREEZER,
+            MieleAppliance.STEAM_OVEN_MK2,
+            MieleAppliance.HOB_INDUCT_EXTR,
+        ),
+        description=MieleBinarySensorDescription(
+            key="state_smart_grid",
+            value_fn=lambda value: value.state_smart_grid,
+            translation_key="smart_grid",
+            entity_category=EntityCategory.DIAGNOSTIC,
+        ),
+    ),
+    MieleBinarySensorDefinition(
+        types=(
+            MieleAppliance.WASHING_MACHINE,
+            MieleAppliance.WASHING_MACHINE_SEMI_PROFESSIONAL,
+            MieleAppliance.TUMBLE_DRYER,
+            MieleAppliance.TUMBLE_DRYER_SEMI_PROFESSIONAL,
+            MieleAppliance.DISHWASHER,
+            MieleAppliance.DISH_WARMER,
+            MieleAppliance.OVEN,
+            MieleAppliance.OVEN_MICROWAVE,
+            MieleAppliance.STEAM_OVEN,
+            MieleAppliance.MICROWAVE,
+            MieleAppliance.COFFEE_SYSTEM,
+            MieleAppliance.HOOD,
+            MieleAppliance.FRIDGE,
+            MieleAppliance.FREEZER,
+            MieleAppliance.FRIDGE_FREEZER,
+            MieleAppliance.ROBOT_VACUUM_CLEANER,
+            MieleAppliance.WASHER_DRYER,
+            MieleAppliance.STEAM_OVEN_COMBI,
+            MieleAppliance.WINE_CABINET,
+            MieleAppliance.WINE_CONDITIONING_UNIT,
+            MieleAppliance.WINE_STORAGE_CONDITIONING_UNIT,
+            MieleAppliance.STEAM_OVEN_MICRO,
+            MieleAppliance.DIALOG_OVEN,
+            MieleAppliance.WINE_CABINET_FREEZER,
+            MieleAppliance.STEAM_OVEN_MK2,
+            MieleAppliance.HOB_INDUCT_EXTR,
+        ),
+        description=MieleBinarySensorDescription(
+            key="state_mobile_start",
+            value_fn=lambda value: value.state_mobile_start,
+            translation_key="mobile_start",
+            entity_category=EntityCategory.DIAGNOSTIC,
+        ),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: MieleConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the binary sensor platform."""
+    coordinator = config_entry.runtime_data
+
+    async_add_entities(
+        MieleBinarySensor(coordinator, device_id, definition.description)
+        for device_id, device in coordinator.data.devices.items()
+        for definition in BINARY_SENSOR_TYPES
+        if device.device_type in definition.types
+    )
+
+
+class MieleBinarySensor(MieleEntity, BinarySensorEntity):
+    """Representation of a Binary Sensor."""
+
+    entity_description: MieleBinarySensorDescription
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return the state of the binary sensor."""
+        return cast(bool, self.entity_description.value_fn(self.device))
```

Comment:
```suggestion
    def is_on(self) -> bool:
        """Return the state of the binary sensor."""
        return self.entity_description.value_fn(self.device)
```

Reply:
According to developer docs is_on should be of type `bool | None`. That way Unknown can be displayed if a None is delivered by the API.

Reply:
yes, but if you never return `None` you can just return `bool` here

Reply:
The aim in the library when it comes to simple values is to return what it gets from the API. So None should be valid.